### ISSUE TITLE
Question regarding opening animation completion

### ISSUE THREAD

**Issue by Steve-Wall** [LeoNatan/LNPopupController#264]
#### Description
I have a question:

When I tap the small player, is there a way to know when the opening of that player has finished? Basically, I want to tap on the small player, and only after it finishes opening completely, do I want to do an action. For example, after tapping the small player, and the player opens up completely, then I want to hide the status bar. 

Please see picture below for example:

![opening](https://user-images.githubusercontent.com/38816624/40088392-ebb5a750-5874-11e8-84c4-192efb0c7144.jpg)

**Comment by LeoNatan** [comment]
Is your only use case hiding the status bar? In that case, use the available mechanisms. The framework supports them.

If it's a general question, there is no way to listen to popup state changes. You can listen to viewWillAppear/viewDidAppear of you content controller to know when the transition has started and completed.

**Comment by Steve-Wall** [comment]
Thanks so much for quick response!

Currently, my only concern is hiding the status bar, but there could be others. When you said `The framework supports them`, do you mean your library supports this - meaning the hiding/showing of status bar?

**Comment by LeoNatan** [comment]
If you use the modern view controller way to hide the status bar, meaning implementing `prefersStatusBarHidden` to return `NO`/`false`, the framework will behave as expected, showing and hiding the status bar when appropriate (for example, in an interactive popup dismiss, the status bar will hide only after the user pulls the popup enough:

![](https://github.com/LeoNatan/LNPopupController/raw/master/Supplements/statusbar_hidden.gif)

This is an old gif, it should behave even smoother now.

**Comment by Steve-Wall** [comment]
Thanks for showing gif - I have exactly what's shown there now actually : ) The problem is that, notice the navigation bar shrink as the popup is appearing. From my research, if we hide the status bar "before" the popup (or any modal VC) appears completely, then you will see the navigation bar shrink.

The solution, I think, is waiting for the popup bar to present complete, and then fading out the status bar. YouTube does this for example. Notice that the status bar only fades out after the player is fully presented.

Youtube example gif:

![status-bar](https://user-images.githubusercontent.com/38816624/40089326-2fb7eb9e-5879-11e8-97d6-ec66175b31bb.gif)

**Comment by LeoNatan** [comment]
I tend to prefer the approach seen in my gif, since when animated it appears as part of the presentation transition. I don't think I will be changing the way this is implement.
If it doesn't animate for you, make sure to return an animation in `preferredStatusBarUpdateAnimation`.

**Comment by Steve-Wall** [comment]
Do you have an example app for that animation? What I just noticed is that mine is not exactly like the gif - instead that status bar hides without the nice animation (this causes a jump on the screen)

I checked your sample project and don't see it : (

**Comment by LeoNatan** [comment]
Change `prefersStatusBarHidden` to return `YES` always and you’ll see it. It’s probably because you don’t have the othe method defined, to tell the system you want an animation.

**Comment by LeoNatan** [comment]
Hmm, your ode looks fine. Try to reproduce on the example app.

**Comment by Steve-Wall** [comment]
Ok, I'll try example app : )

**Comment by Steve-Wall** [comment]
I tried messing around with library but I am not well-versed in Objective-C. Overall, the sample app does not fade out the status bar when the popup appears. The only thing that happens is that the status bar changes color. 

I'm wondering if we hide the status bar in the sample app if it will animate correctly? Could you please look into this?

**Comment by LeoNatan** [comment]
Sounds like a bug, I’ll check.

**Comment by LeoNatan** [comment]
I see the problem reproducing, but it seems like an iOS bug. Not sure yet.

**Comment by Steve-Wall** [comment]
@iDevelopper Thanks, I'll try that out.

@LeoNatan Thanks for looking into this! Would be nice to close out this issue quickly as this is a bad experience - having the screen jump is not pleasant. Please keep us updated

**Comment by LeoNatan** [comment]
Found a solution

**Comment by LeoNatan** [comment]
Please use the latest release.

**Comment by Steve-Wall** [comment]
@LeoNatan Just tried it out in my app. Things got worse : (

Now when I call `presentPopupBar(withContentViewController`, a transparent contentViewController is presented and the app freezes. The tab bar disappears and I can see the batter icon in the status bar. Basically, I am stuck in the screen that presented the popup however I don't see the contentViewController at all - I'm stuck on the screen that presented it. Lastly, the screen is a bit blurry.

**Comment by LeoNatan** [comment]
What

Can you please open a new issue and attach a screenshot? Does this issue reproduce for you in the example app?

**Comment by LeoNatan** [comment]
Unable to reproduce on iOS 10, iOS 11 (8 Plus, X)

**Comment by Steve-Wall** [comment]
@LeoNatan I created a new issues as you requested : ) I put a video to show issue
